libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5170,10 +5158,27 @@ void GCS_MAVLINK::convert_COMMAND_LONG_to_COMMAND_INT(const mavlink_command_long
     out.param2 = in.param2;
     out.param3 = in.param3;
     out.param4 = in.param4;
-    const bool stores_location = command_long_stores_location((MAV_CMD)in.command);
-    out.x = convert_COMMAND_LONG_loc_param(in.param5, stores_location);
-    out.y = convert_COMMAND_LONG_loc_param(in.param6, stores_location);
+
+    // we can't convert NaN/INF to an integer:
+    if (!isfinite(in.param5) || !isfinite(in.param6)) {
+        return MAV_RESULT_DENIED;
+    }
+
+    // check parameters are valid latitude/longitude:
+    if (!check_lat(in.param5) || !check_lng(in.param6)) {
```

Comment:
Don't we only want to do this if the command stores in a location? Possibly that it is also in a global frame, These param are not always lat/lon.

Reply:
Well, that was a brain outage.

Fixed, thanks!